### ISSUE TITLE
[TASK]:

### ISSUE THREAD

**Issue by elaheoveisi** [iHuman-Lab/rescue-analysis#6]
> **Workflow:** Fill out Sections 1, 2, and 3. Wait for PI alignment/approval before creating a branch or writing code.

## 1. Scope & Architecture
* **Objective:** *Determine whether the current 12-participant pilot sample is large enough to reliably detect a real difference in gaze entropy (SGE/GTE).*
* **Config Updates:** List any new parameters needed in the central config file.
* **Entry Point Impacts:** What needs to change in `main.py` to route this task? One import + one skip_run("skip", "power_analysis") block added to main.py, defaulted off, no change to existing step ordering or behavior.

## 2. Implementation Blueprint & LLM Guardrails
* **Logical Steps:**

-  Reuse existing best-run SGE/GTE computation (victim_aoi.run_object_aoi) — unmodified, imported only.
-  Attach condition/expertise labels from config (build_entropy_dataset).
-  Average each participant's two llm rows (gemini, openai) into one paired value per participant vs. their single no_llm (dummy) row.
-  Compute Cohen's dz, power at N=12, and required N for 80% power via statsmodels.stats.power.TTestPower (two-sided, α=0.05).
-  Save outputs, wire into main.py behind a flag.
```makefile
load_participants(): Loads participant IDs and expertise levels from the configuration file.
build_entropy_dataset(): Calculates SGE and GTE for each participant and trial, then adds condition and expertise labels.
paired_power_summary(): Compares the LLM and no-LLM conditions using paired differences and calculates effect size, statistical power, and required sample size.
run(): Runs the complete analysis and saves the entropy features and power-analysis results as CSV files.
TTestPower().power(): Calculates the statistical power for the current sample size.
TTestPower().solve_power(): Estimates how many participants are needed to achieve 80% power.
``` 

* **LLM Boundary:** What *specific* function/syntax are you delegating? *(No structural copy-pasting).* I only used two statistical functions from a library: TTestPower().power() to calculate the likelihood of detecting an effect, and .solve_power() to estimate the required number of participants. For the eye-tracking calculations and AOI labeling, I reused the methods already available in your existing files instead of rewriting them

## 3. Validation & Adversarial Justification
* **The Validation Method:** *(The exact command, script, or test to execute to check correctness).* Manual analysis
* **The Success Condition:** *(What specific, measurable output proves it worked? e.g., falsifiable metrics, tensor shapes).* Two things had to be true: (1) the output table had exactly 24 rows (12 people × 2 conditions) 
* **Justification (Why is this sufficient?):** *Defend your verification method. Address these specific targets:*
    * How does this validation cover the failure space (not just the obvious successes)?
    * What is a case where this validation could pass, but the system is still wrong? How do we catch that?  The checks confirm that the code runs and the calculations are correct, but they cannot identify poor-quality input data. Runs with unusual values, such as exact zeros, should be manually reviewed and possibly excluded before trusting the results.

**Comment by HemuManju** [comment]
Are you studying if there is any change in gaze entropy between No LLM and LLM conditions?

**Comment by elaheoveisi** [comment]
Yes, for each person, we compare how their SGE and GTE scores changed between using an LLM (average of OpenAI and Gemini) and not using it, then check whether we tested enough people to trust that difference. When I combine Gemini and OpenAI, I need at least 27 participants; when I rely only on OpenAI, it increases to 44!!

**Comment by HemuManju** [comment]
Ok, so the power analysis if between no LLM and LLM condition using SGE and GTE features. What was the p-value you got when you compared the eye features between no LLM and LLM condition? Also figure out why leaving Gemini make the number of participants shoot up to 44!

Also are they hyper-parameters for calculating the SGE and GTE features and if yes, then how do we know those parameters are correct for our experiment.

**Comment by elaheoveisi** [comment]
I fixed OpenAI; it is 78, not 44. I added these tables now. I will search for the validity of the hyperparameters for our study, and I will make a new comment.
Main comparison: LLM (Gemini + OpenAI blended) vs. no-LLM (dummy), across all 12 participants:
Metric | Mean difference | p-value
-- | -- | --
SGE | +0.254 | p = 0.0001 — highly significant
GTE | +0.104 | p = 0.076 — not significant (just above 0.05)     


Gemini alone has a stronger true effect on GTE (+0.138) So it changes the results significantly!


What it is | Value
-- | --
Gemini only | avg across 12 people, Gemini vs dummy | 0.138
OpenAI only | avg across 12 people, OpenAI vs dummy | 0.071
Gemini + OpenAI blended | avg of (Gemini, OpenAI) per person, then avg across 12 people | 0.104


Comparison | Metric | Mean diff | p-value | Effect size (dz) | Power at N=12 | N needed for 80% power
-- | -- | -- | -- | -- | -- | --
Gemini + OpenAI (blended) | SGE | +0.254 | 0.0001  | 1.68 | 100% | 6
Gemini + OpenAI (blended) | GTE | +0.104 | 0.076  | 0.57 | 43% | 27
OpenAI only | SGE | +0.226 | 0.0006  | 1.37 | 99% | 7
OpenAI only | GTE | +0.071 | 0.287  | 0.32 | 18% | 78
Gemini only | SGE | +0.281 | 0.0009  | 1.29 | 98% | 7
Gemini only | GTE | +0.138 | 0.025  | 0.75 | 66% | 17



Trial | Saved victims (avg) | Total reward (avg)
-- | -- | --
No-LLM (dummy) | 28.3 | −191.7
Gemini | 31.9 | 163.3
OpenAI | 36.8 | 139.2


Gemini and OpenAI may give advice in different ways. For example, one model may give longer answers or advice that is harder to understand. Because of this, participants may look back and forth between the chat and the game more often.
Participants may also trust the two models differently. If Gemini’s advice feels less reliable, they may check the game more carefully before following its suggestions.
The difference may also come from the order of the experiment, not from the AI model. For example, if participants used one model later, they may have been more tired or more familiar with the task, which could change how they looked at the game and the chat.

**Comment by HemuManju** [comment]
Something is off with GTE and Gemini. The participants do not know which model they are interacting with. Only we know. So I do not agree with the statement: Participants may also trust the two models differently.

If they are looking again and again, does it mean it is better? So in other words, are we interpreting the eye features correctly? 

Regarding the order, we randomised the order. So it shouldn't have any effect.

**Comment by elaheoveisi** [comment]
Their recommendations might be different; also, they are random, but we don't have an equal number of starting orders with Gemini and OpenAI. I am looking to revise the structure of how we decided to pick the AOIs. I found some papers, and I am working to see how they have done this. The structure should be the cause of the difference. I will check and update the code, the issue, and the results.

**Comment by HemuManju** [comment]
I agree their recommendations are different, but that doesn't explain why GTE is different not SGE.

**Comment by elaheoveisi** [comment]
One individual OpenAI session had an unusually low GTE value compared to that same participant's other attempts, and with only 12 participants, a single unusual session can meaningfully shift the result.

**Comment by HemuManju** [comment]
Let's remove the participant and run the analysis.

**Comment by elaheoveisi** [comment]
AOI definitions should have a structure, which I chose to be content-based according to the recommendations of this paper (It was not consistent in the past) [https://www.sciencedirect.com/science/article/pii/S0149763418303075](url)

Category | Contains
-- | --
Victim | victim + fake_victim
Hazard | lava
Access | door + key
Interface | info_panel + chat_panel

<br class="Apple-interchange-newline">
</body>
</html>

For the best run(highest number of victims):

Metric | Comparison | N | Mean diff | p-value | Effect size (dz) | Power at N=12 | N needed for 80% power
-- | -- | -- | -- | -- | -- | -- | --
SGE | Gemini+OpenAI combined | 12 | −0.196 | 0.0057  | −0.99 | 88% | 11
SGE | OpenAI only | 12 | −0.192 | 0.0087  | −0.92 | 83% | 12
SGE | Gemini only | 12 | −0.199 | 0.0063  | −0.97 | 86% | 11
GTE | Gemini+OpenAI combined | 12 | −0.278 | 0.0002  | −1.54 | 100% | 6
GTE | OpenAI only | 12 | −0.252 | 0.0015  | −1.21 | 97% | 8
GTE | Gemini only | 12 | −0.304 | 0.0001  | −1.72 | 100% | 5

<br class="Apple-interchange-newline">
</body>
</html>


I also tried this situation (I preferr this one):  
 Accumulated (all runs pooled/summed per condition, not averaged, not best-run-only):</strong></p>
Metric | Comparison | N | Mean diff | p-value | Effect size (dz) | Power at N=12 | N needed for 80% power
-- | -- | -- | -- | -- | -- | -- | --
GTE | Gemini only | 12 | −0.267 | 0.0006  | −1.37 | 99% | 7
GTE | Combined | 12 | −0.263 | 0.0006  | −1.37 | 99% | 7
GTE | OpenAI only | 12 | −0.258 | 0.0017  | −1.19 | 96% | 8
SGE | Combined | 12 | −0.182 | 0.0084  | −0.92 | 83% | 12
SGE | OpenAI only | 12 | −0.190 | 0.0090  | −0.91 | 82% | 12
SGE | Gemini only | 12 | −0.173 | 0.0143  | −0.84 | 75% | 14


</body>
</html>
